lexers/zig.lua

```diff
@@ -2,88 +2,114 @@
-	'cmpxchgStrong', 'cmpxchgWeak', 'compileError', 'compileLog', 'ctz', 'cUndef', 'divExact',
-	'divFloor', 'divTrunc', 'embedFile', 'enumToInt', 'errorName', 'errorReturnTrace', 'errorToInt',
-	'errSetCast', 'export', 'fence', 'field', 'fieldParentPtr', 'floatCast', 'floatToInt', 'frame',
-	'Frame', 'frameAddress', 'frameSize', 'hasDecl', 'hasField', 'import', 'intCast', 'intToEnum',
-	'intToError', 'intToFloat', 'intToPtr', 'memcpy', 'memset', 'wasmMemorySize', 'wasmMemoryGrow',
-	'mod', 'mulWithOverflow', 'panic', 'popCount', 'ptrCast', 'ptrToInt', 'rem', 'returnAddress',
-	'setAlignStack', 'setCold', 'setEvalBranchQuota', 'setFloatMode', 'setRuntimeSafety', 'shlExact',
-	'shlWithOverflow', 'shrExact', 'shuffle', 'sizeOf', 'splat', 'reduce', 'src', 'sqrt', 'sin',
-	'cos', 'exp', 'exp2', 'log', 'log2', 'log10', 'fabs', 'floor', 'ceil', 'trunc', 'round',
-	'subWithOverflow', 'tagName', 'TagType', 'This', 'truncate', 'Type', 'typeInfo', 'typeName',
-	'TypeOf', 'unionInit'
-}))
+lex:add_rule('function',
+	lex:tag(lexer.FUNCTION_BUILTIN, '@' * lex:word_match(lexer.FUNCTION_BUILTIN)))
+
+-- Bare functions (function calls without @).
+lex:add_rule('function_call',
+	lex:tag(lexer.FUNCTION, lexer.word * #(lexer.space^0 * '(')))
+
+-- Struct methods (word followed by dot and another word).
+lex:add_rule('method', lex:tag(lexer.FUNCTION, lexer.word * '.' * lexer.word))
 
 -- Strings.
-local sq_str = P('L')^-1 * lexer.range("'", true)
-local dq_str = P('L')^-1 * lexer.range('"', true)
-lex:add_rule('string', token(lexer.STRING, sq_str + dq_str))
+local raw_str = P('\\\\') * lexer.range('\n', false, false) -- Multi-line strings
+local sq_str = lexer.range("'", true) -- Character/Byte literal
+local dq_str = lexer.range('"', true) -- String/Byte array literal
+lex:add_rule('string', lex:tag(lexer.STRING, raw_str + sq_str + dq_str))
 
 -- Identifiers.
-lex:add_rule('identifier', token(lexer.IDENTIFIER, lexer.word))
+lex:add_rule('identifier', lex:tag(lexer.IDENTIFIER, lexer.word))
 
 -- Comments.
-local doc_comment = lexer.to_eol('///', true)
-local comment = lexer.to_eol('//', true)
-lex:add_rule('comment', token(lexer.COMMENT, doc_comment + comment))
+local doc_comment = lexer.to_eol('///', true) -- Documentation comments
+local comment = lexer.to_eol('//', true)     -- Single-line comments
+lex:add_rule('comment', lex:tag(lexer.COMMENT, doc_comment + comment))
 
 -- Numbers.
-lex:add_rule('number', token(lexer.NUMBER, lexer.number))
+local binary = P('0b') * S('01')^1
+local octal = P('0o') * S('01234567')^1  
+local hex = P('0x') * S('0123456789abcdefABCDEF')^1
+local decimal = lexer.number
+lex:add_rule('number', lex:tag(lexer.NUMBER, binary + octal + hex + decimal))
 
 -- Operators.
-lex:add_rule('operator', token(lexer.OPERATOR, '..' + S('+-/*%<>!=^&|?~:;,.()[]{}')))
+lex:add_rule('operator', lex:tag(lexer.OPERATOR, S('+-/*%<>!=^&|?~:;,.()[]{}') +
+	P('**') + P('<<') + P('>>') + P('==') + P('!=') + P('<=') + P('>=')))
```

Comment:
None of these will ever be matched. The first `S()` pattern will always match these. I honestly don't see a need for these two-character strings.